worker.js

```diff
@@ -63,40 +85,51 @@ var getBucketList = function (cb) {
 var readBuckets = function (tasks) {
   console.info('--- Started indexing all buckets ---');
   async.parallel(tasks,
-  function (err, results) {
-    if (err) {
-      return console.error(err);
-    }
-    console.info('--- Finished indexing all buckets ---');
-    // Get image, sensor, and provider counts and save to analytics collection
-    return Promise.all([
-      Meta.count(),
-      Meta.distinct('properties.sensor'),
-      Meta.distinct('provider')
-    ]).then(function (res) {
-      var counts = {};
-      counts.image_count = res[0];
-      counts.sensor_count = res[1].length;
-      counts.provider_count = res[2].length;
-      analytics.addAnalyticsRecord(counts, function (err) {
-        // Catch error in record addition
+    // Results is an [[tasks]]
+    function (err, results) {
+      if (err) {
+        return console.error(err);
+      }
+      results = _.flatten(results);
+      results = results.map(function (task) {
+        return async.retryable(task);
+      });
+      async.parallelLimit(results, 5, function (err, results) {
         if (err) {
-          console.error(err);
+          db.close();
+          return console.error(err);
         }
-        console.info('--- Added new analytics record ---');
-        db.close();
+        console.info('--- Finished indexing all buckets ---');
+        // Get image, sensor, and provider counts and save to analytics collection
+        return Promise.all([
+          Meta.count(),
+          Meta.distinct('properties.sensor'),
+          Meta.distinct('provider')
+        ]).then(function (res) {
+          var counts = {};
+          counts.image_count = res[0];
+          counts.sensor_count = res[1].length;
+          counts.provider_count = res[2].length;
+          analytics.addAnalyticsRecord(counts, function (err) {
+            // Catch error in record addition
+            if (err) {
+              console.error(err);
+            }
+            console.info('--- Added new analytics record ---');
+            db.close();
+          });
+          // Catch error in db query promises
+        }).catch(function (err) {
+          db.close();
```

Comment:
Stick this in a `.then()` after the `.catch()` (effectively `finally` to consistently close the db in one place)?